pr_agent/git_providers/bitbucket_provider.py

```diff
@@ -72,24 +74,38 @@
         post_parameters_list = []
         for suggestion in code_suggestions:
             body = suggestion["body"]
+            original_suggestion = suggestion.get('original_suggestion', None)  # needed for diff code
+            if original_suggestion:
+                try:
+                    existing_code = original_suggestion['existing_code'].rstrip() + "\n"
+                    improved_code = original_suggestion['improved_code'].rstrip() + "\n"
+                    diff = difflib.unified_diff(existing_code.split('\n'),
+                                                improved_code.split('\n'), n=999)
+                    patch_orig = "\n".join(diff)
+                    patch = "\n".join(patch_orig.splitlines()[5:]).strip('\n')
+                    diff_code = f"\n\n```diff\n{patch.rstrip()}\n```"
+                    # replace ```suggestion ... ``` with diff_code, using regex:
+                    body = re.sub(r'```suggestion.*?```', diff_code, body, flags=re.DOTALL)
```

Comment:
**Suggestion:** Replace the regex pattern in `re.sub` to ensure it correctly targets only the intended markdown fenced code blocks. [general, importance: 5]
```suggestion
                    body = re.sub(r'```suggestion[\s\S]*?```', diff_code, body, flags=re.DOTALL)
```